crates/runtime/src/traphandlers/unix.rs

```diff
@@ -158,6 +158,19 @@ unsafe fn get_pc(cx: *mut libc::c_void, _signum: libc::c_int) -> *const u8 {
         } else if #[cfg(all(target_os = "freebsd", target_arch = "x86_64"))] {
             let cx = &*(cx as *const libc::ucontext_t);
             cx.uc_mcontext.mc_rip as *const u8
+        } else if #[cfg(all(target_os = "openbsd", target_arch = "x86_64"))] {
+            let cx = cx as *const u64;
+            // TODO: get Rust's `libc` crate to include a definition for
+            // OpenBSD's `ucontext_t`.  Until then, we hardcode this offset.
+            // `ucontext_t` is defined in `<sys/signal.h>` as a typedef for
+            // `struct sigcontext`, which is defined in `<machine/signal.h>`.
+            // The `sc_rip` field is a u64 and is preceded by 21 u64 fields.
+            // This is the kernel-entry register-save frame format and is
+            // *unlikely* to change across OpenBSD versions, but note that
+            // OpenBSD makes no guarantee of ABI compatibility across any
+            // release.
```

Comment:
Sure thing -- created rust-lang/libc#2242. It might be a bit until this is easily usable without building rustc from source (the platform doesn't have rustup so it would likely be available with the packages in the next OpenBSD release in October) but then this PR isn't in any hurry either :-)

Reply:
So it looks like there was an ongoing PR for this (rust-lang/libc#2189) that is somewhat complex, and stalled. For now I added a C helper and I'm happy to revisit this when Rust support exists, if that seems reasonable to you! (Also I misspoke above w.r.t. waiting for an OS package update -- `libc` is just a crate, of course, not part of the stdlib distro proper.)

Reply:
The C wrapper looks fine to me. I suggest adding a link to rust-lang/libc#2189 in a comment so that people looking to refactor that code can easily check on its status.